pinot-common/src/main/java/org/apache/pinot/common/utils/LoggerUtils.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.utils;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.Configuration;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+
+
+/**
+ * Logger utils for process level logger management.
+ */
+public class LoggerUtils {
+  private static final String ROOT = "root";
+  private static final String NAME = "name";
+  private static final String LEVEL = "level";
+  private static final String FILTER = "filter";
+
+  private LoggerUtils() {
+  }
+
+  public static Map<String, Object> setLoggerLevel(String loggerName, String logLevel) {
+    LoggerContext context = (LoggerContext) LogManager.getContext(false);
+    Configuration config = context.getConfiguration();
+    LoggerConfig loggerConfig =
+        loggerName.equalsIgnoreCase(ROOT) ? config.getRootLogger() : config.getLoggerConfig(loggerName);
+    loggerConfig.setLevel(Level.valueOf(logLevel));
+    // This causes all Loggers to re-fetch information from their LoggerConfig.
+    context.updateLoggers();
+    return getLoggerResponse(loggerConfig);
+  }
+
+  public static Map<String, Object> getLoggerInfo(String loggerName) {
+    LoggerContext context = (LoggerContext) LogManager.getContext(false);
+    Configuration config = context.getConfiguration();
+    LoggerConfig loggerConfig =
+        loggerName.equalsIgnoreCase(ROOT) ? config.getRootLogger() : config.getLoggerConfig(loggerName);
```

Comment:
Will `getLoggerConfig()` actually return null if it doesn't get a match? The documentation says "This will remove tokens from the package name as necessary or return the root LoggerConfig if no other matches were found".

Reply:
It won't. So we just add a logger name check before here.